applications/sdp/mspi/src/main.c

```diff
@@ -52,215 +57,494 @@ typedef struct __packed {
+
+static struct xfer_io_mode_cfg get_io_modes(enum mspi_io_mode mode)
+{
+	struct xfer_io_mode_cfg xfer_mode;
+
+	switch (nrfe_mspi_dev_cfg.io_mode) {
+	case MSPI_IO_MODE_SINGLE:
+		xfer_mode.command = BASE_IO_MODE_SINGLE;
+		xfer_mode.address = BASE_IO_MODE_SINGLE;
+		xfer_mode.data = BASE_IO_MODE_SINGLE;
+		break;
+	case MSPI_IO_MODE_DUAL:
+		xfer_mode.command = BASE_IO_MODE_DUAL;
+		xfer_mode.address = BASE_IO_MODE_DUAL;
+		xfer_mode.data = BASE_IO_MODE_DUAL;
+		break;
+	case MSPI_IO_MODE_DUAL_1_1_2:
+		xfer_mode.command = BASE_IO_MODE_SINGLE;
+		xfer_mode.address = BASE_IO_MODE_SINGLE;
+		xfer_mode.data = BASE_IO_MODE_DUAL;
+		break;
+	case MSPI_IO_MODE_DUAL_1_2_2:
+		xfer_mode.command = BASE_IO_MODE_SINGLE;
+		xfer_mode.address = BASE_IO_MODE_DUAL;
+		xfer_mode.data = BASE_IO_MODE_DUAL;
+		break;
+	case MSPI_IO_MODE_QUAD:
+		xfer_mode.command = BASE_IO_MODE_QUAD;
+		xfer_mode.address = BASE_IO_MODE_QUAD;
+		xfer_mode.data = BASE_IO_MODE_QUAD;
+		break;
+	case MSPI_IO_MODE_QUAD_1_1_4:
+		xfer_mode.command = BASE_IO_MODE_SINGLE;
+		xfer_mode.address = BASE_IO_MODE_SINGLE;
+		xfer_mode.data = BASE_IO_MODE_QUAD;
+		break;
+	case MSPI_IO_MODE_QUAD_1_4_4:
+		xfer_mode.command = BASE_IO_MODE_SINGLE;
+		xfer_mode.address = BASE_IO_MODE_QUAD;
+		xfer_mode.data = BASE_IO_MODE_QUAD;
+		break;
+	case MSPI_IO_MODE_OCTAL:
+		xfer_mode.command = BASE_IO_MODE_OCTAL;
+		xfer_mode.address = BASE_IO_MODE_OCTAL;
+		xfer_mode.data = BASE_IO_MODE_OCTAL;
+		break;
+	case MSPI_IO_MODE_OCTAL_1_1_8:
+		xfer_mode.command = BASE_IO_MODE_SINGLE;
+		xfer_mode.address = BASE_IO_MODE_SINGLE;
+		xfer_mode.data = BASE_IO_MODE_OCTAL;
+		break;
+	case MSPI_IO_MODE_OCTAL_1_8_8:
+		xfer_mode.command = BASE_IO_MODE_SINGLE;
+		xfer_mode.address = BASE_IO_MODE_OCTAL;
+		xfer_mode.data = BASE_IO_MODE_OCTAL;
+		break;
+	case MSPI_IO_MODE_HEX:
+	case MSPI_IO_MODE_HEX_8_8_16:
+	case MSPI_IO_MODE_HEX_8_16_16:
+	case MSPI_IO_MODE_MAX:
+	default:
+		/* TODO: Jira ticket: NRFX-6875 error*/
+		break;
+	}
 
-static void process_packet(const void *data, size_t len, void *priv)
+	return xfer_mode;
+}
+
+static void ep_bound(void *priv)
+{
+	atomic_set_bit(&ipc_atomic_sem, NRFE_MSPI_EP_BOUNDED);
+}
+
+static void ep_recv(const void *data, size_t len, void *priv)
 {
 	(void)priv;
-	(void)len;
-	nrfe_mspi_flpr_response_t response;
-	uint8_t opcode = *(uint8_t *)data;
 
-	response.opcode = opcode;
+	process_packet(data, len);
+}
 
-	switch (opcode) {
-	case NRFE_MSPI_CONFIG_PINS: {
-		/* TODO: Process pinctrl config data
-		 * nrfe_mspi_pinctrl_soc_pin_t *pins_cfg = (nrfe_mspi_pinctrl_soc_pin_t *)data;
-		 * response.opcode = pins_cfg->opcode;
-		 *
-		 * for (uint8_t i = 0; i < NRFE_MSPI_PINS_MAX; i++) {
-		 *         uint32_t psel = NRF_GET_PIN(pins_cfg->pin[i]);
-		 *         uint32_t fun = NRF_GET_FUN(pins_cfg->pin[i]);
-		 *         NRF_GPIO_Type *reg = nrf_gpio_pin_port_decode(&psel);
-		 * }
-		 */
+static void ll_prepare_transfer(enum base_io_mode xfer_mode, uint32_t data_length)
```

Comment:
This should be a part of HRT?

Reply:
yes, IMO this should be part of HRT. This function has direct influence on transfer timings.

Reply:
no, it is low level, but it is only configuration there is nothing timing critical going on there